src/main/java/org/opensearch/knn/indices/ModelMetadata.java

```diff
@@ -351,12 +371,37 @@ public static ModelMetadata fromString(String modelMetadataString) {
                 description,
                 error,
                 trainingNodeAssignment,
-                methodComponentContext
+                methodComponentContext,
+                VectorDataType.FLOAT
+            );
+        } else if (modelMetadataArray.length == 10) {
```

Comment:
This if -else is going to increase and it's not a good practice to use this , Can we use ReGistry Pattern to re-factor this , I will leave up to you to think how to avoid this if -else condition.

Reply:
Simple Example of how can we avoid all duplications code ,,,

```
public static ModelMetadata fromString(String modelMetadataString) {
    String[] modelMetadataArray = modelMetadataString.split(DELIMITER, -1);
    int length = modelMetadataArray.length;

    if (length < 7 || length > 9) {
        // Throw Exception 
    }

    KNNEngine knnEngine = KNNEngine.getEngine(modelMetadataArray[0]);
    SpaceType spaceType = SpaceType.getSpace(modelMetadataArray[1]);
    int dimension = Integer.parseInt(modelMetadataArray[2]);
    ModelState modelState = ModelState.getModelState(modelMetadataArray[3]);
    String timestamp = modelMetadataArray[4];
    String description = modelMetadataArray[5];
    String error = modelMetadataArray[6];

    String trainingNodeAssignment = "";
    MethodComponentContext methodComponentContext = MethodComponentContext.EMPTY;

    if (length >= 8) {
        trainingNodeAssignment = modelMetadataArray[7];
    }
    if (length == 9) {
        methodComponentContext = MethodComponentContext.fromClusterStateString(modelMetadataArray[8]);
    }
    return new ModelMetadata(
        knnEngine,
        spaceType,
        dimension,
        modelState,
        timestamp,
        description,
        error,
        trainingNodeAssignment,
        methodComponentContext
    );
}
```

Reply:
refactored this method